tarun_kumar_PostgreAssignment/src/controller/mark-controller.ts

```diff
@@ -0,0 +1,42 @@
+import { Context } from 'vm';
+import * as subjectService from '../services/subject-service';
+import * as markService from '../services/mark-service';
+
+
+export async function addMarks(ctx: Context) {
+    try {
+        const requestData = ctx.request.body;
+
+        const student_id: string = requestData.student_id;
+        const subject_name: string = requestData.subject_name.toLowerCase();
+        const marks: number = requestData.marks;
+        const subject_id: string = await subjectService.getSubjectId(subject_name);
+
+        await markService.addMark(student_id, subject_id, marks);
+
+        ctx.body = {
+            message: `marks for student with ${student_id} is added`,
+        };
+    } catch (e) {
```

Comment:
can we ready the error message and see what issue and then send message accordingly. Foreign key constraint, primary key constraint, any other issue